ksql-engine/src/main/java/io/confluent/ksql/query/id/HybridQueryIdGenerator.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.query.id;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Contains both a Sequential Query Id Generator and a Specific QueryId Generator
+ * Can toggle between the two depending on which one is currently activated
+ * The query id generation method was changed from using
+ */
+public class HybridQueryIdGenerator implements QueryIdGenerator {
+
+  private final SequentialQueryIdGenerator legacyGenerator;
+  private final SpecificQueryIdGenerator newGenerator;
+  private QueryIdGenerator activeGenerator;
```

Comment:
does this need to be thread safe? I think the overhead of setting it so might be low enough to be worth just doing it (e.g. use an `AtomicReference` here)